internal/stack/resources.go

```diff
@@ -24,8 +24,8 @@ import (
 var static embed.FS
 
 const (
-	// SnapshotFile is the docker-compose snapshot.yml file name.
-	SnapshotFile = "snapshot.yml"
+	// ComposeFile is the docker compose file.
+	ComposeFile = "docker-compose.yml"
```

Comment:
There is an issue here but not sure if it would be too complicated to add support for this scenario(s).

In one of my tests I faced one error:
1. Elastic stack running with a previous elastic-package version (using snapshot.yaml). 
2. Install elastic-package with the contents of this branch.
3. Try to run `elastic-package stack down -v` command failed:
    ```bash
     $ elastic-package stack down -v 
    2024/05/07 15:40:54 DEBUG Enable verbose logging
    2024/05/07 15:40:54 DEBUG Distribution built without a version tag, can't determine release chronology. Please consider using official releases at https://github.com/elastic/elastic-package/releases
    Take down the Elastic stack
    Error: tearing down the stack failed: stopping docker containers failed: could not create docker compose project: could not find Docker Compose configuration file: /home/mariorodriguez/.elastic-package/profiles/default/stack/docker-compose.yml: stat /home/mariorodriguez/.elastic-package/profiles/default/stack/docker-compose.yml: no such file or directory
    ``` 


Same error happens with these sequence of commands:
1. Tear down Elastic stack with previous version `elastic-package stack down`
2. Install elastic-package with the contents of this branch.
3. Run again `elastic-package stack down -v`
    - `elastic-package` does not find the new `docker-compose.yml` because `elastic-package stack up` has not run yet.

Reply:
Just checked that`snapshot.yml` file is kept in `~/.elastic-package/profiles/default/stack/snapshot.yml` even if the new file is created and used by elastic-package.

I guess there should be any issue, depending on the elastic-package version one or other file would be used.

Reply:
Solved by adding "migrations" in https://github.com/elastic/elastic-package/pull/1820/commits/5bfb5dfe032dfc6c15cb5bde89cad78cfcfe14a1.
I will split the PR in two because it is starting to have too many things.

Reply:
Refactors reverted in new PR https://github.com/elastic/elastic-package/pull/1824.